cmd/client/wallet/main.go

```diff
@@ -213,27 +235,73 @@ func createWalletNode() *node.Node {
 	return w
 }
 
+func getPassphrase(prompt string) string {
+	fmt.Printf(prompt)
+	bytePassword, err := terminal.ReadPassword(int(syscall.Stdin))
+	if err != nil {
+		panic("read password error")
+	}
+	password := string(bytePassword)
+	fmt.Println()
+	return password
+}
+
 func processNewCommnad() {
-	randomBytes := [32]byte{}
-	_, err := io.ReadFull(cRand.Reader, randomBytes[:])
+	newCommand.Parse(os.Args[2:])
+	noPass := *newCommandNoPassPtr
+	password := ""
 
-	if err != nil {
-		fmt.Println("Failed to get randomness for the private key...")
-		return
+	if !noPass {
+		fmt.Printf("Passphrase: ")
+		bytePassword, err := terminal.ReadPassword(int(syscall.Stdin))
+		if err != nil {
+			panic("read password error")
+		}
+		password = string(bytePassword)
+		fmt.Println()
 	}
-	priKey, err := crypto2.GenerateKey()
+
+	account, err := ks.NewAccount(password)
 	if err != nil {
-		panic("Failed to generate the private key")
+		fmt.Printf("new account error: %v\n", err)
 	}
-	storePrivateKey(crypto2.FromECDSA(priKey))
-	fmt.Printf("New account created with address:{%s}\n", crypto2.PubkeyToAddress(priKey.PublicKey).Hex())
-	fmt.Printf("Please keep a copy of the private key:{%s}\n", hex.EncodeToString(crypto2.FromECDSA(priKey)))
+	fmt.Printf("account: %s\n", account.Address.Hex())
+	fmt.Printf("URL: %s\n", account.URL)
 }
 
 func processListCommand() {
-	for i, key := range readPrivateKeys() {
-		fmt.Printf("Account %d:{%s}\n", i, crypto2.PubkeyToAddress(key.PublicKey).Hex())
-		fmt.Printf("    PrivateKey:{%s}\n", hex.EncodeToString(key.D.Bytes()))
+	listCommand.Parse(os.Args[2:])
+	noPass := *listCommandNoPassPtr
+
+	accounts := ks.Accounts()
+	for _, account := range accounts {
+		fmt.Printf("account: %s\n", account.Address.Hex())
+		fmt.Printf("URL: %s\n", account.URL)
+		password := ""
+		newpass := ""
+		if !noPass {
+			password = getPassphrase("Passphrase: ")
+			newpass = getPassphrase("Export Passphrase: ")
```

Comment:
What is export passphrase used for?

Reply:
you can have a different passphrase when you export the keystore.  That's a function in eth keystore.